packages/nx-plugin/src/plugin/target/configuration-target.ts

```diff
@@ -1,12 +1,22 @@
 import { TargetConfiguration } from '@nx/devkit';
 import { RunCommandsOptions } from 'nx/src/executors/run-commands/run-commands.impl';
 import { PACKAGE_NAME } from '../../internal/constants';
+import { CP_TARGET_NAME } from '../constants';
 
-export function createConfigurationTarget(
-  projectName?: string,
-): TargetConfiguration<RunCommandsOptions> {
+export function createConfigurationTarget(options?: {
+  targetName?: string;
+  projectName?: string;
+  bin?: string;
+}): TargetConfiguration<RunCommandsOptions> {
+  const {
+    projectName,
+    bin = PACKAGE_NAME,
+    targetName = CP_TARGET_NAME,
+  } = options ?? {};
   const projectFlag = projectName && ` --project=${projectName}`;
   return {
-    command: `nx g ${PACKAGE_NAME}:configuration${projectFlag ?? ''}`,
+    command: `nx g ${bin}:configuration --skipTarget --targetName=${targetName}${
+      projectFlag ?? ''
+    }`,
   };
```

Comment:
Why not use the `objectToCliArgs` util here? Seems like a good fit because you have some logic for conditional arguments.

Also, if the target name isn't set, then maybe we don't have to explicitly set the default value?

Reply:
Refactored. Good hint!

Reply:
for the target name, we need a string as target name to set it in the project. In this case it is a bit useless as we also use `skipTarget`. I just did not want to wrestle with the scheme of the configuration generator.